ExomeDepth/igv_xml_session.py

```diff
@@ -1,59 +1,333 @@
-    new_session = "{0}_{1}_{2}_igv.xml".format(sample_id, statistic, args.runid)
-    igv_ed_hc_test = "{0}_{1}_test".format(igv_ed_hc, statistic)
-    igv_ed_umcu_test = "{0}_{1}_test".format(igv_ed_umcu, statistic)
-    bam_coverage = "{0}_coverage".format(bam)
-    bam_junctions = "{0}_junctions".format(bam)
-    min_axis, mid_axis, max_axis = axis
-    ratioid_UMCU = "{0}_UMCU".format(statistic)
-    ratioid_HC = "{0}_HC".format(statistic)
-    substitute_dic = {'session_var':new_session, 'igv_ed_umcu':igv_ed_umcu, 'igv_ed_hc':igv_ed_hc, 
-                    'bam':bam, 'vcf_hc':vcf_hc, 'sample_id':sample_id, 'igv_ed_hc_test':igv_ed_hc_test, 
-                    'igv_ed_umcu_test':igv_ed_umcu_test, 'bam_coverage':bam_coverage, 'bam_junctions':bam_junctions,  
-                    'vcf_SNV':vcf_SNV, 'min_axis':min_axis, 'mid_axis':mid_axis, 'max_axis':max_axis, 
-                    'ratioid_UMCU':ratioid_UMCU, 'ratioid_HC':ratioid_HC 
-                   }
+
+def parse_ped_family(ped_file, sampleid):
+    sample_dic = {}
+    for line in ped_file:
+        splitline = line.split()
+        if splitline[1] not in sample_dic:
+            sample_dic[splitline[1]] = [splitline[0], splitline[2], splitline[3]]
+    """ Return familyID, child, father, mother """
+    return sample_dic[sampleid][0], sampleid, sample_dic[sampleid][1], sample_dic[sampleid][2]
+
+
+def make_single_igvsession(args, sample_id, statistic, igv_extension, vcf_extension, bam):
+    template_file = Template(open(args.template_xml).read())
+
+    """ Data files XML variables"""
+    if args.pipeline == "iap":  # For analysis based on IAP
+        bam_path = "../{0}/mapping/{1}".format(args.sampleid, bam)  # BAM path
+        snv_vcf = "../single_sample_vcf/{0}.filtered_variants.vcf".format(args.sampleid)  # SNV VCF file
+    elif args.pipeline == "nf":  # For NF pipeline.
+        bam_path = "../bam_files/{0}".format(bam)  # BAM path
+        snv_vcf = "../single_sample_vcf/{0}_{1}.vcf".format(args.sampleid, args.runid)  # SNV VCF file
+
+    hc_cnv_vcf = "HC/HC_{0}_{1}_{2}_{3}".format(args.refset, args.sampleid, args.runid, vcf_extension)  # HC CNV-VCF file
+    baf = "../baf/{0}_baf.igv".format(args.sampleid)  # BAF file
+    igv_hc_ratio = "igv_tracks/HC_{0}_{1}_{2}_{3}".format(
+        args.refset, args.sampleid, args.runid, igv_extension
+    )  # CNV igv session with ratios for HC
+    igv_umcu_ratio = "igv_tracks/UMCU_{0}_{1}_{2}_{3}".format(
+        args.refset, args.sampleid, args.runid, igv_extension
+    )  # CNV igv session with ratios for UMCU
+
+    """ ID XML variables"""
+    session = "{0}_{1}_{2}_igv.xml".format(sample_id, statistic, args.runid)  # Session ID
+    hc_cnv_vcf_id = "CNV:HC_{0}_{1}_{2}_{3}".format(
+        args.refset, args.sampleid, args.runid, vcf_extension
+    )  # HC CNV -VCF track id in IGV
+    snv_vcf_id = "SNV/MNV:{0}".format(snv_vcf.split("/")[-1])  # SNV/MNV track ID in IGV
```

Comment:
You are using this pattern of getting the file name a lot -> maybe create a separate function and reuse.